cloudinit/util.py

```diff
@@ -3210,6 +3210,18 @@ def _compare_version(self, other: "Version") -> int:
         return -1
 
 
+def should_log_deprecation(version: str) -> bool:
```

Comment:
this could be used too (simplify this logic too](https://github.com/canonical/cloud-init/blob/5ce2ee36e243b4cc178fef28bc0da06212ac7607/cloudinit/config/schema.py#L800)

if we're making a helper, could we re-use it there too?

Reply:
I'm not sure I understand what that particular `if` is trying to convey. It has more stuff than `should_log_deprecations()`, but I'm not sure why.

Reply:
@TheRealFalcon I think it just means we want to invert the conditional to append to deprecations if should_log_deprecation else to info_deprecations:
```diff
--- a/cloudinit/config/schema.py
+++ b/cloudinit/config/schema.py
@@ -43,6 +43,7 @@ from cloudinit.util import (
     get_modules_from_dir,
     load_text_file,
     load_yaml,
+    should_log_deprecation,
     write_file,
 )
 
@@ -795,16 +796,12 @@ def validate_cloudconfig_schema(
         if isinstance(
             schema_error, SchemaDeprecationError
         ):  # pylint: disable=W1116
-            if "devel" != features.DEPRECATION_INFO_BOUNDARY and (
-                schema_error.version == "devel"
-                or Version.from_str(schema_error.version)
-                > Version.from_str(features.DEPRECATION_INFO_BOUNDARY)
-            ):
+            if should_log_deprecation(schema_error.version):
+                deprecations.append(SchemaProblem(path, schema_error.message))
+            else:
                 info_deprecations.append(
                     SchemaProblem(path, schema_error.message)
                 )
-            else:
-                deprecations.append(SchemaProblem(path, schema_error.message))
         else:
             errors.append(SchemaProblem(path, schema_error.message))
 

```

Reply:
With the above diff, downstream unittests pass on jammy with all quilt patches applied (which pins deprecation_info_boundary)